### PR TITLE
Prepare for modules with multiple input/output pins

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

File: tools/topology/topology2/include/common/tokens.conf [hunk 1]
```diff
@@ -22,6 +22,8 @@ Object.Base.VendorToken {
 		# the widget name that the queue is connected from/to.
 		sink_pin_binding_wname	413
 		src_pin_binding_wname	414
+		num_input_audio_formats	415
+		num_output_audio_formats	416
```
Comment (plbossart) [inline tools/topology/topology2/include/common/tokens.conf:26]: num_sink_audio_formats and num_source_audio_formats for consistency?

File: tools/topology/topology2/include/components/cavs/src_format.conf [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #src format array
 			num_audio_formats 42
+			num_input_audio_formats 42
+			num_output_audio_formats 42
```
Comment (plbossart) [inline tools/topology/topology2/include/components/cavs/src_format.conf:4]: the answer is always 42.

File: tools/topology/topology2/include/common/tokens.conf [hunk 3]
```diff
@@ -162,6 +162,7 @@ Object.Base.VendorToken {
 		in_interleaving_style	1906
 		in_fmt_cfg		1907
 		in_sample_type		1908
+		pin_index		1909
```
Comment (plbossart) [inline tools/topology/topology2/include/common/tokens.conf:165]: does this belong in this PR? I don't see the link with input/output

Comment (ranj063) [reply tools/topology/topology2/include/common/tokens.conf:165]: It's a token needed for the pin_index attributes in the new classes

**Comment by dbaluta** [comment]
@ranj063 Care to explain a little bit better what is an **output Pin**? An output pin is a data line of a DAI?

**Comment by ranj063** [comment]
> @ranj063 Care to explain a little bit better what is an **output Pin**? An output pin is a data line of a DAI?

@dbaluta this is purely from a module pov. A module could have multiple input and output modules connected to it. The term pin refers to the connection between modules and is used to specify the audio formats required at each end of the connection.

**Review by kv2019i — APPROVED** [review]

**Review by juimonen — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

File: tools/topology/topology2/include/common/sink_audio_format.conf [hunk 4]
```diff
@@ -0,0 +1,137 @@
+# Class definition for sink pin audio format object
```
Comment (ujfalusi) [inline tools/topology/topology2/include/common/sink_audio_format.conf:1]: I'm confused.
Don't we have support for pins other than index 0? I would thin that they somehow work and somehow they actually get audio format, no?

Isn't the problem lies that we have only one set of audio format definition, which is global for the whole module and for all of it's pins?

Comment (ranj063) [reply tools/topology/topology2/include/common/sink_audio_format.conf:1]: @ujfalusi we dont. The only audio formats we use today are for pin 0 and always intput and output together in pairs. Clearly this wont work

**Review by ujfalusi — APPROVED** [review]

**Comment by kv2019i** [comment]
One timeout in https://sof-ci.01.org/sofpr/PR7014/build3661/devicetest/index.html , not related to this PR, proceeding with merge.

**Pull request merged**